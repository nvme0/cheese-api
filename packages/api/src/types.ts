export interface Cheese {
  id: string;
  name: string;
  imageUrl?: string;
  color: string;
  pricePerKilo: number;
}
