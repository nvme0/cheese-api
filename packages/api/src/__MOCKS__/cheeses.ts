import { Cheese } from "../types";

export const organicFeta: Cheese = {
  id: "9ac773e1-c953-4ba8-90d3-02e0eeddef44",
  name: "Organic Feta",
  imageUrl: "https://cheese.com/media/img/cheese/Feta_.jpg",
  color: "white",
  pricePerKilo: 549,
};

export const somersetBrie: Cheese = {
  id: "3ac413d1-36e5-43fd-a050-785a522cd940",
  name: "Somerset Brie",
  imageUrl: "https://cheese.com/media/img/cheese/brie.jpg",
  color: "white",
  pricePerKilo: 325,
};

export const mascarpone: Cheese = {
  id: "c56777da-0d78-4bad-ad35-57c546ce6387",
  name: "Mascarpone (Australian)",
  imageUrl: "https://cheese.com/media/img/cheese/1186-mascarpone_australian.jpg",
  color: "ivory",
  pricePerKilo: 199,
};

export const cottageCheese: Cheese = {
  id: "a2553e96-05e7-48bd-ab8b-8c70350be2f0",
  name: "Cottage Cheese (Australian)",
  imageUrl: "https://cheese.com/media/img/cheese/Cottagecheese.jpg",
  color: "white",
  pricePerKilo: 399,
};

export const dryJack: Cheese = {
  id: "f411ab0a-306e-47e0-a7ef-31c4b4ede802",
  name: "Dry Jack",
  imageUrl: "https://cheese.com/media/img/cheese/427px-Sonoma_Dry_Jack.jpg",
  color: "pale yellow",
  pricePerKilo: 249,
};

export default [organicFeta, somersetBrie, mascarpone, cottageCheese, dryJack];
