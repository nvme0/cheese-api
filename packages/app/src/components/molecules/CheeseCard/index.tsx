import React from "react";
import { Card, CardActionArea, CardContent, CardMedia, Grid, Typography } from "@material-ui/core";

import useStyles from "./styles";
import centsToDollars from "@utils/centsToDollars";

export interface Props {
  title: string;
  imageUrl?: string;
  color: string;
  pricePerKilo: number;
}

const CheeseCard = (props: Props) => {
  const { title, imageUrl, color, pricePerKilo } = props;
  const classes = useStyles();

  const dollarsPerKilo = centsToDollars(pricePerKilo);

  return (
    <Card>
      <CardActionArea>
        <CardMedia className={classes.media} image={imageUrl} title={title} />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {title}
          </Typography>
          <Grid container>
            <Grid item xs={12}></Grid>
          </Grid>
        </CardContent>
      </CardActionArea>
      <CardContent>
        <Grid container>
          <Grid item xs={12}>
            <Typography variant="body1" component="p">
              Color: {color}
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography variant="body1" component="p">
              Price Per Kilo: ${dollarsPerKilo}
            </Typography>
          </Grid>
        </Grid>
      </CardContent>
    </Card>
  );
};

export default CheeseCard;
