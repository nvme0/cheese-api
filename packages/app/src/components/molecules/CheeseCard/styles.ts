import { makeStyles, useTheme } from "@material-ui/core/styles";

export const appBarHeight = 58;
export const footerHeight = 48;

const styles = makeStyles((theme) => ({
  media: {
    height: 140,
  },
}));

const useStyles = () => {
  const theme = useTheme();
  return styles(theme);
};

export default useStyles;
