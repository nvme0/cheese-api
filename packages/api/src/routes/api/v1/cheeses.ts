import { Router } from "express";
import cheeses from "../../../__MOCKS__/cheeses";

const router = Router();

router.get("/", (req, res) => {
  res.send(cheeses);
});

router.get("/:id", (req, res) => {
  const {
    params: { id },
  } = req;

  const requestedCheese = cheeses.find((cheese) => cheese.id === id);
  if (requestedCheese) {
    res.send(requestedCheese);
  } else {
    res.sendStatus(404);
  }
});

export default router;
