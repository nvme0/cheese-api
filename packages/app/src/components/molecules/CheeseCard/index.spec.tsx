import React from "react";
import { render, screen } from "@testing-library/react";

import CheeseCard from ".";
import { organicFeta } from "../../../__MOCKS__/cheeses";
import { Cheese } from "@app/types";
import centsToDollars from "@utils/centsToDollars";

const renderCheeseCard = (cheese: Cheese) => {
  const { name, color, imageUrl, pricePerKilo } = cheese;
  return render(<CheeseCard title={name} color={color} imageUrl={imageUrl} pricePerKilo={pricePerKilo} />);
};

describe("CheeseCard", () => {
  it("renders correctly", () => {
    const component = renderCheeseCard(organicFeta);
    expect(component).toMatchSnapshot();
  });

  it("contains a title", () => {
    const { name } = organicFeta;
    renderCheeseCard(organicFeta);
    const title = screen.getByText(name);
    expect(title).toBeInTheDocument();
  });

  it("contains the color", () => {
    const { color } = organicFeta;
    renderCheeseCard(organicFeta);
    const title = screen.getByText(`Color: ${color}`);
    expect(title).toBeInTheDocument();
  });

  it("contains the price per kilo", () => {
    const { pricePerKilo } = organicFeta;
    renderCheeseCard(organicFeta);
    const title = screen.getByText(`Price Per Kilo: $${centsToDollars(pricePerKilo)}`);
    expect(title).toBeInTheDocument();
  });
});
