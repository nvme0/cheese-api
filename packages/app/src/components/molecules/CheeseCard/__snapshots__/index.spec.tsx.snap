// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheeseCard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-1"
            style="background-image: url(https://cheese.com/media/img/cheese/Feta_.jpg);"
            title="Organic Feta"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Organic Feta
            </h2>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Color: 
                white
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Price Per Kilo: $
                5.49
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-1"
          style="background-image: url(https://cheese.com/media/img/cheese/Feta_.jpg);"
          title="Organic Feta"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Organic Feta
          </h2>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            />
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Color: 
              white
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Price Per Kilo: $
              5.49
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
